### ISSUE TITLE
Issue saving basic task to disk via api

### ISSUE THREAD

**Issue by gaieges** [Flexget/Flexget#1383]
### Expected behaviour:

Post to `/api/tasks/` using the below curl command responds back with a 201

### Actual behaviour:

Post to `/api/tasks/` using the below curl command responds back with: 500
API Response:

```
{
    "message": "Internal Server Error"
}
```

### Steps to reproduce:
- Install flexget with config below (before)
- Start daemon & get token
- Run following command with token:

```bash
curl localhost:3539/api/tasks/ -XPOST -H 'content-type: application/json' -H 'Authorization: Token TOKEN' -d '{ "name":"myNewTask",  "config": {     "manual": true,    "rss": {      "url": "http://google.com"    }  }}'
```

#### Config:

Config file before (works fine):
```
web_server:
  bind: 0.0.0.0
  port: 3539
api:    True
webui:  True

schedules:
- interval: {minutes: 15}
  tasks: '*'


tasks:
  noop:
    manual: true
    rss:
        url: 'http://google-old.com'
```

Config file after failure: **empty**

  
#### Log:

```
2016-08-31 19:22 ERROR    flexget.api                   Exception on /tasks/ [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1639, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1625, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python2.7/dist-packages/flask_restful/__init__.py", line 477, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flask/views.py", line 84, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flask_restplus/resource.py", line 42, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/utils/database.py", line 34, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/api/__init__.py", line 143, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/flexget/api/tasks.py", line 92, in post
    self.manager.save_config()
  File "/usr/local/lib/python2.7/dist-packages/flexget/manager.py", line 630, in save_config
    config_file.write(yaml.dump(self.user_config, default_flow_style=False))
TypeError: write() argument 1 must be unicode, not str
```

### Additional information:

- Flexget Version: 2.3.14
- Python Version: Python 2.7.12
- Installation method: pip
- OS and version: Ubuntu 16.04.1 LTS

Please verify that the following data is present before submitting your issue:

**Comment by liiight** [comment]
Will be fixed when #1389 is merged.

**Comment by liiight** [comment]
This should be fixed in v2.4.0. Please verify and let me know.
